src/PointwiseFunctions/AnalyticSolutions/ForceFree/AlfvenWave.cpp

```diff
@@ -0,0 +1,134 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/AnalyticSolutions/ForceFree/AlfvenWave.hpp"
+
+#include <cstddef>
+#include <memory>
+#include <pup.h>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "PointwiseFunctions/InitialDataUtilities/InitialData.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace ForceFree::Solutions {
+
+AlfvenWave::AlfvenWave(const double wave_speed) : wave_speed_(wave_speed) {
+  ASSERT(abs(wave_speed_) < 1.0,
+         "The wave speed cannot be exactly -1.0 or 1.0");
+  lorentz_factor_ = 1.0 / sqrt(1.0 - square(wave_speed_));
+}
+
+AlfvenWave::AlfvenWave(CkMigrateMessage* msg) : InitialData(msg) {}
+
+std::unique_ptr<evolution::initial_data::InitialData> AlfvenWave::get_clone()
+    const {
+  return std::make_unique<AlfvenWave>(*this);
+}
+
+void AlfvenWave::pup(PUP::er& p) {
+  InitialData::pup(p);
+  p | wave_speed_;
+  p | lorentz_factor_;
+  p | background_spacetime_;
+}
+
+PUP::able::PUP_ID AlfvenWave::my_PUP_ID = 0;
+
+DataVector AlfvenWave::wave_profile(const DataVector& x_prime) {
+  // Compute the B_z'(=-E_x') at the rest frame of the wave
+  auto result = make_with_value<DataVector>(x_prime, 1.0);
+  for (size_t i = 0; i < result.size(); ++i) {
+    const double& xi = x_prime[i];
+    if (xi > -0.1) {
+      if (xi <= 0.1) {
+        result[i] = 1.15 + 0.15 * sin(5 * M_PI * xi);
+      } else {
+        result[i] = 1.3;
+      }
+    }
+  }
+  return result;
+}
+
+DataVector AlfvenWave::charge_density(const DataVector& x_prime) {
+  // Compute the charge density q = Div(E) at the rest frame of the wave
+  auto result = make_with_value<DataVector>(x_prime, 0.0);
+  for (size_t i = 0; i < result.size(); ++i) {
+    const double& xi = x_prime[i];
+    if ((xi > -0.1) and (xi <= 0.1)) {
+      result[i] = -0.75 * M_PI * cos(5.0 * M_PI * xi);
+    }
+  }
+  return result;
+}
+
+tuples::TaggedTuple<Tags::TildeE> AlfvenWave::variables(
+    const tnsr::I<DataVector, 3>& x, double t,
+    tmpl::list<Tags::TildeE> /*meta*/) const {
+  const auto& x_coord = x.get(0);
```

Comment:
Prefer `get<N>(x)` over `x.get(N)` when `N` is a constant.  Change the rest of the occurrences as well.